api/controllers/UserController.js

```diff
@@ -75,6 +75,40 @@ module.exports = _.mapValues({
     await req.user.deleteAccount();
     return res.redirect('/');
   },
+  async changePassword (req, res) {
+    const params = req.allParams();
+    Validation.requireParams(params, ['oldPassword', 'newPassword']);
+    const oldPassport = await Passport.findOne({user: req.user.name, protocol: 'local'});
+    const validate = oldPassport.validatePassword.bind(oldPassport);
+    const isValid = await Promise.promisify(validate)(params.oldPassword);
+    if (!isValid) {
+      // If the provided oldPassword is incorrect, stop immediately.
+      return res.forbidden('Incorrect password');
+    }
+    // Otherwise, attempt to create a new Passport for the user with the new password.
+    // This is done *before* deleting the old Passport; otherwise, the user will be locked out of their
+    // account if there's an error creating the new Passport.
+    const newPassport = await Passport.create({
+      protocol: 'local',
+      user: req.user.name,
+      password: params.newPassword,
+      accessToken: require('crypto').randomBytes(48).toString('base64')
```

Comment:
Yeah, I think this is the right way to do things, it looks the same as how they set up the registration, so, looks good to me.